### DISCUSSION TITLE
Attach new tab/pane to existing window?

### DISCUSSION THREAD

**Discussion by lycurgus** [wezterm/wezterm#1017] [Q&A] [answered]
I'd like to be able to run a command outside of my wezterm instance (e.g. via a hotkey or `dmenu`) that, instead of creating a new wezterm window, adds a tab (or a pane) to the existing window instead. Is that possible?

I don't have multiplexing set up, but from what I've seen so far, things like `wezterm cli` seem to be built around interacting with wezterm from within itself - I get a different set of information if I run `wezterm cli list` in wezterm vs in another terminal. Have I missed something?

**Comment by wez** [comment, marked as answer]
Each instance of wezterm listens on a unix domain socket.  Processes spawned by that wezterm have environment variables set that `wezterm cli` can use to locate and talk to that instance:

```bash
; env | grep WEZTERM
..
WEZTERM_PANE=2
WEZTERM_UNIX_SOCKET=/Users/wez/.local/share/wezterm/gui-sock-50325
```

If you know the `WEZTERM_UNIX_SOCKET` for the instance you want, then you could set that explicitly.

There isn't a mechanism that can conveniently enumerate those at this time.

However, if you configure [Multiplexing with a unix domain socket](https://wezfurlong.org/wezterm/multiplexing.html#unix-domains) then you can run a central `wezterm-mux-server` process and then have your wezterm GUI instances all talk to that.  The multiplexer socket location is configured in the config file, so you won't need to mess with the environment.

**Reply by lycurgus** [reply]
Ok, I think I don't need multiplexing, then - I just need to communicate with the existing socket.

(I also ran into the same confusion as #712 when I tried multiplexing - I started a multiplexed session, started another window and it connected to the first... and then when I tried to close the second window via my window manager it wanted to kill the whole multiplexed session along with the tabs I had in the first window. I did succeed in closing the second one via `kill` without it killing the whole session, but I agree a detach operation seems sensible!)

I only need there to be one session, so I think I can just grab `gui-sock-*` and use that to talk via `wezterm cli`.

Is there a way to control the geometry of a tab I spawn with `wezterm cli spawn`? It looks like it wants to open as 80x24 even when the existing window is larger than that; so I end up with a little tab taking up the corner of the window....

**Reply by wez** [reply]
> Is there a way to control the geometry of a tab I spawn with `wezterm cli spawn`? It looks like it wants to open as 80x24 even when the existing window is larger than that; so I end up with a little tab taking up the corner of the window....

Sounds like https://github.com/wez/wezterm/issues/920
